Feature enumerable methods

sjcco opened the pull request:
Micoverse 2nd ruby project

Recreated Enumerable methods

sjcco committed:
add methods 1-8

sjcco committed:
add methods 9 and 10

sjcco committed:
fixed linter error

sjcco committed:
fixed linter error

sjcco committed:
fixed linter error

BrianSammit reviewed (changes requested):
CHANGES REQUIRED!! :recycle: :red_circle: 
---------------
Hello team!! :wave: you have an amazing project :1st_place_medal: :muscle: 
However, there are a few changes to make :hammer_and_wrench: see them below :point_down: 
--------------

- [ ] Remember to add a Readme file, Add a descriptive README file to your repositories.
- [ ] Set up your `development` branch in GitHub as a default branch in your case should be `dev` branch. [Here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) is a reminder on how to work with Git flow. 
- [ ] Your pull request should be `feature-enumerable-methods` to `dev` branch, not to `main` branch. 
- [ ] #my_all? when no block or argument is given returns true when none of the collection members are false or nil. Instead is returning False. 
- [ ] #my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class. Instead is returning True. 
- [ ] #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer. Instead is returning True. 
- [ ] #my_map returns an Enumerator if no block is given.
- [ ] #my_inject searches for the longest word in an array of strings. I got this 'undefined local variable or method `i`'
- [ ] #my_inject does not mutate the original array. I got this 'undefined local variable or method `i`'
- [ ] #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method. I got this 'undefined local variable or method `param2`'
- [ ] #multiply_els accepts an array as an argument and multiplies all the elements of the array together using #my_inject. I got this 'undefined local variable or method `param2`'
--------------------
Fix the required changes :point_up: and send it for another review :muscle: :+1: 
HAPPY CODDING !!

![200 (1)](https://user-images.githubusercontent.com/64173189/107793340-2b70e000-6d24-11eb-9aef-1e5d36da5a54.gif)

sjcco committed:
fixed variable names

sjcco committed:
fixed see notes

#my_all? when no block or argument is given returns true when none of the collection members are false or nil. Instead is returning False.

 #my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class. Instead is returning True.

 #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer. Instead is returning True.

#my_map returns an Enumerator if no block is given.

#my_inject searches for the longest word in an array of strings. I got this 'undefined local variable or method i'

sjcco committed:
fixed linter error

sjcco committed:
add readme file

sjcco committed:
add rubocop config file

sjcco committed:
Update enumerables.rb
